File: js/1744.js
const fs = require('fs');

const inputs = fs.readFileSync('example.txt').toString().trim().split('\n');

const N = +inputs[0];
const arr = [];
const arr2 = [];
let ans = 0;

for (let i = 1; i <= N; i++) {
  const value = +inputs[i];
  if (value > 0) arr.push(value);
  else arr2.push(value);
}

arr.sort((a, b) => b - a);
arr2.sort((a, b) => a - b);

for (let i = 0; i < arr.length; i++) {
  if (i === arr.length - 1) {
    ans += arr[i];
  } else {
    const a = arr[i];
    const b = arr[i + 1];

    if (a * b > a + b) {
      ans += a * b;
      i++;
    } else {
      ans += a;
    }
  }
}

for (let i = 0; i < arr2.length; i++) {
  if (i === arr2.length - 1) {
    ans += arr2[i];
  } else {
    const a = arr2[i];
    const b = arr2[i + 1];

    if (a * b > a + b) {
      ans += a * b;
      i++;
    } else {
      ans += a;
    }
  }
}

console.log(ans);

File: c++/1744.cpp
#include <algorithm>
#include <iostream>
#include <vector>

using namespace std;

vector<int> positive;
vector<int> negative;
int ans;

int main() {
  freopen("example.txt", "r", stdin);

  int N;
  cin >> N;

  while (N--) {
    int a;
    cin >> a;

    if (a > 0)
      positive.push_back(a);
    else
      negative.push_back(a);
  }

  sort(positive.begin(), positive.end(), greater<>());
  sort(negative.begin(), negative.end());

  int pSize = positive.size();
  int nSize = negative.size();

  for (int i = 0; i < pSize; i++) {
    if (i == pSize - 1) {
      ans += positive[i];
    } else {
      int a = positive[i];
      int b = positive[i + 1];

      if (a * b > a + b) {
        ans += a * b;
        i++;
      } else {
        ans += a;
      }
    }
  }

  for (int i = 0; i < nSize; i++) {
    if (i == nSize - 1) {
      ans += negative[i];
    } else {
      int a = negative[i];
      int b = negative[i + 1];

      if (a * b > a + b) {
        ans += a * b;
        i++;
      } else {
        ans += a;
      }
    }
  }

  cout << ans;

  return 0;
}